tests/testthat/test_regr_xgboost.R

```diff
@@ -15,18 +15,18 @@ test_that("regr_xgboost", {
     if (is.null(parset$verbose)) parset$verbose = 0L
     if (is.null(parset$nrounds)) parset$nrounds = 1L
     if (is.null(parset$objective)) parset$objective = "reg:linear"
-    pars = list(data = data.matrix(regr.train[,1:13]), label = as.numeric(regr.train[,14]))
+    pars = list(data = data.matrix(regr.train[,1:3]), label = as.numeric(regr.train[,14]))
     pars = c(pars, parset)
     set.seed(getOption("mlr.debug.seed"))
     model = do.call(xgboost::xgboost, pars)
-    #model = xgboost::xgboost(data = data.matrix(regr.train[,1:13]), verbose = 0L,
+    #model = xgboost::xgboost(data = data.matrix(regr.train[,1:3]), verbose = 0L,
     #label = as.numeric(regr.train[,14]),
     #nrounds = 20, objective = "reg:linear", missing = NULL)
-    old.predicts.list[[i]] = predict(model, data.matrix(regr.test[,1:13]))
+    old.predicts.list[[i]] = predict(model, data.matrix(regr.test[,1:3]))
   }
   
   #set.seed(getOption("mlr.debug.seed"))
-  testSimpleParsets("regr.xgboost", regr.df, regr.target, regr.train.inds,
+  testSimpleParsets("regr.xgboost", regr.df[, c(1:3, 14)], regr.target, regr.train.inds,
```

Comment:
We should then use `regr.num.df` here and open up an issue for the other learners (also maybe for the other classif learners).